[CS] Code style Tabs must be used to indent lines; round 2

photodude opened the pull request:
Pull Request for Issue Tabs must be used to indent lines; spaces are not allowed.

### Summary of Changes
- Tabs must be used to indent lines; spaces are not allowed

Working through Automatic fixers to remove cases where we have had rule exceptions

[Automatically fixed with Joomla code standards 2.0.0 PHPCS2-alpha2 fixers](https://github.com/joomla/coding-standards/releases/tag/2.0.0-alpha2)

None of the manual only fixes have been applied

### Testing Instructions
Merge by code review

### Expected result
code style has been applied as listed above, old code style testing on drone does not error.

### Actual result
code style had not been applied. [Autofixers from the Joomla code standards 2.0.0 PHPCS2 alpha2](https://github.com/joomla/coding-standards/releases/tag/2.0.0-alpha2) were used to implement fixable code style

### Documentation Changes Required
none

photodude committed:
[CS] Code style Tabs must be used to indent lines; round 2

- Tabs must be used to indent lines; spaces are not allowed

photodude committed:
fix some indenting not fixed by the auto fixer

Quy commented on templates/beez3/html/com_contact/category/default_children.php:
```diff
@@ -32,15 +32,15 @@
 				<?php echo JText::_('COM_CONTACT_CAT_NUM'); ?></dt>
 				<dd><?php echo $child->numitems; ?></dd>
 			</dl>
 		<?php endif; ?>
```
Indent

photodude replied:
fixed

Quy commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -116,9 +116,9 @@
+	</dl>
 <?php endif; ?>
-<?php  if (isset($images->image_intro) and !empty($images->image_intro)) : ?>
+<?php if (isset($images->image_intro) and !empty($images->image_intro)) : ?>
```
change `and`

photodude replied:
See this discussion on use of `and`/`or` vs `&&`/`||` in template mixed HTML/PHP files
there is no consensus on what we allow/disallow here.
https://github.com/joomla/coding-standards/pull/191

Quy replied:
I understand. For this page, we have a mix of `and/or` and `&&/||` so it should be one or the other.

photodude replied:
for now I'll set this file to `and`/`or`, we can always adjust it back using the auto fixers

Quy replied:
My preference is `&&/||`. I still consider it PHP even though in HTML. I don't like that you have think/consider its syntax based on whether it is a PHP or HTML file.

photodude replied:
That's my preference too. Code consistency for the win. but I made a choice here based on the some of the other arguments. 
https://github.com/joomla/coding-standards/pull/191#issuecomment-312276418

photodude replied:
fixed despite the differing opinions on this. Code consistency for the win.

Quy commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -116,9 +116,9 @@
 		</dd>
 <?php endif; ?>
 <?php if ($params->get('show_author') or $params->get('show_category') or $params->get('show_create_date') or $params->get('show_modify_date') or $params->get('show_publish_date') or $params->get('show_parent_category') or $params->get('show_hits')) :?>
```
change `or`

photodude replied:
fixed

Quy commented on templates/beez3/html/com_newsfeeds/categories/default.php:
```diff
@@ -13,28 +13,26 @@
-				<div class="category-desc  base-desc">
-					<?php  echo JHtml::_('content.prepare', $this->parent->description, '', 'com_newsfeeds.categories'); ?>
+		<?php if ($this->params->get('show_base_description')) : ?>
+		<?php // If there is a description in the menu parameters use that; ?>
```
remove `;`

photodude replied:
fixed

Quy commented on templates/beez3/html/com_newsfeeds/categories/default.php:
```diff
@@ -13,28 +13,26 @@
+			<?php else: ?>
+				<?php // Otherwise get one from the database if it exists. ?>
+				<?php if ($this->parent->description) : ?>
+					<div class="category-desc  base-desc">
```
remove extra space

photodude replied:
fixed

Quy commented on templates/beez3/html/com_newsfeeds/category/default_children.php:
```diff
@@ -33,9 +33,9 @@
+			<?php endif; ?>
 
-            <?php if ($this->params->get('show_cat_items') == 1) :?>
+			<?php if ($this->params->get('show_cat_items') == 1) :?>
```
add space after `:`

photodude replied:
fixed

Quy commented on templates/beez3/html/com_newsfeeds/categories/default_items.php:
```diff
@@ -31,7 +31,7 @@
 		<?php endif; ?>
-        <?php endif; ?>
+		<?php endif; ?>
 		<?php if ($this->params->get('show_cat_items_cat') == 1) :?>
```
add space after `:`

photodude replied:
fixed

photodude committed:
Fix some more indenting and spacing

photodude committed:
use and/or in template mixed HTML/PHP files

For consistancy, Until a decision is made on joomla/coding-standards#191
use `and`/`or` in template mixed HTML/PHP files rather than our normal
required `&&`/`||` requierment that we have for full PHP files

photodude committed:
Use the && and || operators

preferences to Use the && and || operators

Quy commented on templates/beez3/html/com_contact/category/default_children.php:
```diff
@@ -32,15 +32,15 @@
-				<?php echo JText::_('COM_CONTACT_CAT_NUM'); ?></dt>
-				<dd><?php echo $child->numitems; ?></dd>
-			</dl>
+			<?php if ($this->params->get('show_cat_items') == 1) :?>
```
Add space after `:`

Quy commented on templates/beez3/html/com_contact/category/default_children.php:
```diff
@@ -32,15 +32,15 @@
+				</dl>
 		<?php endif; ?>
-            <?php if (count($child->getChildren()) > 0 ) :
+			<?php if (count($child->getChildren()) > 0 ) :
```
Remove space after `0`

Quy commented on templates/beez3/html/com_contact/contact/default_form.php:
```diff
@@ -43,30 +43,30 @@
-			          <?php endif ?>
-			     <?php endforeach;?>
+				<?php foreach ($this->form->getFieldsets() as $fieldset) : ?>
+					<?php if ($fieldset->name !== 'contact'):?>
```
Add space before and after `:`

Quy commented on templates/beez3/html/com_contact/contact/default_form.php:
```diff
@@ -43,30 +43,30 @@
-			     <?php endforeach;?>
+				<?php foreach ($this->form->getFieldsets() as $fieldset) : ?>
+					<?php if ($fieldset->name !== 'contact'):?>
+						<?php $fields = $this->form->getFieldset($fieldset->name);?>
```
Add space after `;`

Quy commented on templates/beez3/html/com_contact/contact/default_form.php:
```diff
@@ -43,30 +43,30 @@
+							<div class="control-group">
+								<?php if ($field->hidden) : ?>
+									<div class="controls">
+									 <?php echo $field->input;?>
```
Add space after `;`

Quy commented on templates/beez3/html/com_contact/contact/default_form.php:
```diff
@@ -43,30 +43,30 @@
+									<div class="control-label">
+										<?php echo $field->label; ?>
+										<?php if (!$field->required && $field->type !== 'Spacer') : ?>
+											<span class="optional"><?php echo JText::_('COM_CONTACT_OPTIONAL');?></span>
```
Add space after `;`

Quy commented on templates/beez3/html/com_contact/contact/default_form.php:
```diff
@@ -43,30 +43,30 @@
+											<span class="optional"><?php echo JText::_('COM_CONTACT_OPTIONAL');?></span>
+										<?php endif; ?>
+									</div>
+									<div class="controls"><?php echo $field->input;?></div>
```
Add space after `;`

Quy commented on templates/beez3/html/com_contact/contact/default_form.php:
```diff
@@ -43,30 +43,30 @@
+										<?php endif; ?>
+									</div>
+									<div class="controls"><?php echo $field->input;?></div>
+								<?php endif;?>
```
Same here

Quy commented on templates/beez3/html/com_contact/contact/default_form.php:
```diff
@@ -43,30 +43,30 @@
+									<div class="controls"><?php echo $field->input;?></div>
+								<?php endif;?>
+							</div>
+						<?php endforeach;?>
```
Same here

Quy commented on templates/beez3/html/com_contact/contact/default_form.php:
```diff
@@ -43,30 +43,30 @@
+							</div>
+						<?php endforeach;?>
+					<?php endif ?>
+				<?php endforeach;?>
```
Same here

Quy commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -115,10 +115,10 @@
 <?php endif; ?>
-<?php if ($params->get('show_author') or $params->get('show_category') or $params->get('show_create_date') or $params->get('show_modify_date') or $params->get('show_publish_date') or $params->get('show_parent_category') or $params->get('show_hits')) :?>
- 	</dl>
+<?php if ($params->get('show_author') || $params->get('show_category') || $params->get('show_create_date') || $params->get('show_modify_date') || $params->get('show_publish_date') || $params->get('show_parent_category') || $params->get('show_hits')) :?>
```
Add space  after `:`

photodude replied:
At some point we are deviating far from the scope of this PR. The intention was originally to only focus on fixing the indenting to use Tabs. 

Should we fix all of the other code style items? Sure, at some point. But we should be cautious in how much we add to the scope of this PR.

Quy replied:
You're right. Lets do these in a separate PR.

photodude replied:
Sounds like a plan.
Let's get this and #19350 merged and work towards fixing the rest of the CS items that can be auto fixed done. Then lets come back for the templates and see what can be done.

zero-24 commented:
Merged on review thanks @photodude

photodude commented:
Thanks everyone
